File: backend/app/services/workspace_service.py
from app.repositories.workspace_repository import WorkspaceRepository

class WorkspaceService:
    def __init__(self, db):
        self.workspace_repo = WorkspaceRepository(db)

    def get_workspace(self, workspace_id: int):
        return self.workspace_repo.get_workspace(workspace_id)

    def get_workspaces(self, user_id: int, skip: int = 0, limit: int = 100):
        return self.workspace_repo.get_workspaces(user_id, skip, limit)

    def create_workspace(self, workspace_data: dict):
        return self.workspace_repo.create_workspace(workspace_data)

    def update_workspace(self, workspace_id: int, workspace_data: dict):
        return self.workspace_repo.update_workspace(workspace_id, workspace_data)

    def delete_workspace(self, workspace_id: int):
        return self.workspace_repo.delete_workspace(workspace_id)

File: src/services/workspace.service.js
import api from './api';

export const workspaceService = {
  getWorkspaces: async () => {
    return await api.get('/workspaces');
  },
  getWorkspaceById: async (id) => {
    return await api.get(`/workspaces/${id}`);
  },
  createWorkspace: async (data) => {
    return await api.post('/workspaces', data);
  },
  updateWorkspace: async (id, data) => {
    return await api.put(`/workspaces/${id}`, data);
  },
  deleteWorkspace: async (id) => {
    return await api.delete(`/workspaces/${id}`);
  },
};

export default workspaceService;
